sdk/src/main/java/io/dapr/client/DaprHttpBuilder.java

```diff
@@ -68,38 +55,30 @@ public DaprHttp build(Properties properties) {
    * @return Instance of {@link DaprHttp}
    */
   private DaprHttp buildDaprHttp(Properties properties) {
-    if (OK_HTTP_CLIENT == null) {
+    if (HTTP_CLIENT == null) {
       synchronized (LOCK) {
-        if (OK_HTTP_CLIENT == null) {
-          OkHttpClient.Builder builder = new OkHttpClient.Builder();
-          Duration readTimeout = Duration.ofSeconds(properties.getValue(HTTP_CLIENT_READ_TIMEOUT_SECONDS));
-          builder.readTimeout(readTimeout);
-
-          Dispatcher dispatcher = new Dispatcher();
-          dispatcher.setMaxRequests(properties.getValue(HTTP_CLIENT_MAX_REQUESTS));
-          // The maximum number of requests for each host to execute concurrently.
-          // Default value is 5 in okhttp which is totally UNACCEPTABLE!
-          // For sidecar case, set it the same as maxRequests.
-          dispatcher.setMaxRequestsPerHost(HTTP_CLIENT_MAX_REQUESTS.get());
-          builder.dispatcher(dispatcher);
-
-          ConnectionPool pool = new ConnectionPool(properties.getValue(HTTP_CLIENT_MAX_IDLE_CONNECTIONS),
-                  KEEP_ALIVE_DURATION, TimeUnit.SECONDS);
-          builder.connectionPool(pool);
-
-          OK_HTTP_CLIENT = builder.build();
+        if (HTTP_CLIENT == null) {
+          int maxRequests = properties.getValue(HTTP_CLIENT_MAX_REQUESTS);
+          Executor executor = Executors.newFixedThreadPool(maxRequests);
+          HTTP_CLIENT = HttpClient.newBuilder()
+              .executor(executor)
```

Comment:
When connecting to `localhost` `OkHttpClient` use `HTTP 1.1` so we need to use the same HTTP protocol version.